### PR TITLE
[Chore] Refactoring to MVVM

### PR DESCRIPTION
**Pull request by Kharonus**
- introduction of MVVM design pattern
- view first approach
- dependency injection from app root
- minor removal of unused code

This PR is a complete refactoring with the goal of introducing MVVM design pattern with view first approach. A lot of files were touched, as with MVVM I introduced complete IoC with constructor injection in most viewmodels and services.

### CODE REVIEW

File: src/OpenProject/ViewModels/WebViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    public WebViewModel(BrowserManager browserManager, JavaScriptBridge javaScriptBridge)
+    {
+      _browserManager = browserManager;
+      _javaScriptBridge = javaScriptBridge;
+
+      _browserManager.Initialize();
+      InitializeIpcConnection();
+
+      if (UserSettings.GetBool("checkupdates"))
+        CheckUpdates();
```
Comment (GeorgDangl) [inline src/OpenProject/ViewModels/WebViewModel.cs:30]: I'm always anxious over missing curly braces😀 It reminds me of the Apple Curly Braces bug a few years ago.

**Review by GeorgDangl — COMMENTED** [review]
Pretty cool PR😀

**Comment by Kharonus** [comment]
Thanks :D

File: src/OpenProject/ViewModels/WebViewModel.cs (hunk 1 above)
Comment (Kharonus) [reply src/OpenProject/ViewModels/WebViewModel.cs:30]: I know this anxiety :D For really simple one-liners like a method call or a `return;` I nevertheless like it to drop the curlies for readability's sake.

I know of the C bug with curlys, but not the mentioned apple bug. You have a link?

Comment (GeorgDangl) [reply src/OpenProject/ViewModels/WebViewModel.cs:30]: Essentially this error:
image

Afair, it came from some merge conflict, and disabled SSL validation on millions of devices.

https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/

But there are lots of Google hits for the bug, not sure which article is the best. Just remember it from some years ago.

Comment (Kharonus) [reply src/OpenProject/ViewModels/WebViewModel.cs:30]: Ah, I see. Well, when it comes from merges, this is really fucked up and can only be solved by correct automated tests. When coding directly in an IDE, this cannot happen, as the IDE directly puts a warning on the indentation.

Interesting story though :D as a non-apple user, I didn't even know about this bug.

Comment (GeorgDangl) [reply src/OpenProject/ViewModels/WebViewModel.cs:30]: >this cannot happen

Famous last words😀

Comment (Kharonus) [reply src/OpenProject/ViewModels/WebViewModel.cs:30]: You know, I like it pretentious, just before failing (with grace). :D

**Pull request merged**